packages/react-native/Libraries/promiseRejectionTrackingOptions.js

```diff
@@ -22,7 +24,50 @@ let rejectionTrackingOptions: $NonMaybeType<Parameters<enable>[0]> = {
       // $FlowFixMe[method-unbinding] added when improving typing for this parameters
       message = Error.prototype.toString.call(rejection);
       const error: Error = (rejection: $FlowFixMe);
-      stack = error.stack;
+
+      // Print pretty unhandled rejections while on DEV
+      if (__DEV__) {
+        const parseErrorStack = require('./Core/Devtools/parseErrorStack');
+        const symbolicateStackTrace = require('./Core/Devtools/symbolicateStackTrace');
+        const LogBox = require('./LogBox/LogBox').default;
+
+        const parsedStack = parseErrorStack(error.stack);
+
+        symbolicateStackTrace(parsedStack)
```

Comment:
This shouldn't live here. Logbox can handle this better by first showing the unsymbolicated stacktrace and then replacing the relevant log when the symbolication is ready.